--- linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi/common/cmm_info.c	2024-10-15 18:31:18.563574731 +0800
+++ linux-3.10.108b/drivers/net/wireless/mtk/mt7603/mt7603_wifi/common/cmm_info.c	2024-10-15 18:41:22.644663527 +0800
@@ -2770,7 +2770,11 @@
     ==========================================================================
 */
 #ifndef WH_EZ_SETUP
-#define	LINE_LEN	(4+65+20+23+9+7+7+3)	/* Channel+SSID(2*32+1)+Bssid+Security+Signal+WiressMode+ExtCh+NetworkType*/
+
+// modify by dimmalex for UTF8 code SSID
+#define	LINE_LEN	(4+66+20+23+9+7+7+3)	/* Channel+SSID(2*32+1)+Bssid+Security+Signal+WiressMode+ExtCh+NetworkType*/
+//#define	LINE_LEN	(4+65+20+23+9+7+7+3)	/* Channel+SSID(2*32+1)+Bssid+Security+Signal+WiressMode+ExtCh+NetworkType*/
+
 #endif
 
 #ifdef AIRPLAY_SUPPORT
@@ -2795,7 +2799,11 @@
 	INT         Rssi = 0;
 	UINT        Rssi_Quality = 0;
 	NDIS_802_11_NETWORK_TYPE    wireless_mode;
-	CHAR		Ssid[2*MAX_LEN_OF_SSID + 2];
+
+    // modify by dimmalex for UTF8 code SSID
+	CHAR		Ssid[2*(MAX_LEN_OF_SSID+1) + 1];
+	//CHAR		Ssid[2*MAX_LEN_OF_SSID + 2];
+
 	RTMP_STRING SecurityStr[32] = {0};
 	NDIS_802_11_ENCRYPTION_STATUS	ap_cipher = Ndis802_11EncryptionDisabled;
 	NDIS_802_11_AUTHENTICATION_MODE	ap_auth_mode = Ndis802_11AuthModeOpen;
@@ -2808,7 +2816,10 @@
 
 
 		/*SSID*/
-	NdisZeroMemory(Ssid, (MAX_LEN_OF_SSID +1));
+    // modify by dimmalex for UTF8 code SSID
+	NdisZeroMemory(Ssid, (2*(MAX_LEN_OF_SSID+1)+1));
+	//NdisZeroMemory(Ssid, (MAX_LEN_OF_SSID +1));
+
 	if (RTMPCheckStrPrintAble((PCHAR)pBss->Ssid, pBss->SsidLen))
 		NdisMoveMemory(Ssid, pBss->Ssid, pBss->SsidLen);
 	else
@@ -2823,7 +2834,11 @@
 		for (idx = 0; (idx < MAX_LEN_OF_SSID) && (idx < pBss->SsidLen); idx++)
 			sprintf(Ssid + 2 + (idx*2), "%02X", (UCHAR)pBss->Ssid[idx]);
 	}
-	sprintf(msg+strlen(msg), "%-33s", Ssid);
+
+    // modify by dimmalex for UTF8 code SSID
+	sprintf(msg+strlen(msg), "%-66s", Ssid);
+	//sprintf(msg+strlen(msg), "%-33s", Ssid);
+
 #ifdef AIRPLAY_SUPPORT
 	/* IsUniCode SSID */
     if (isUniCodeSsid == TRUE)
@@ -3136,11 +3151,21 @@
 	sprintf(msg + strlen(msg), "Total=%-4d", pAdapter->ScanTab.BssNr);
 	sprintf(msg + strlen(msg), "%s", "\n");
 #ifdef AIRPLAY_SUPPORT
-		sprintf(msg+strlen(msg),"%-4s%-33s%-4s%-20s%-23s%-9s%-7s%-7s%-3s\n",
+
+    // modify by dimmalex for UTF8 code SSID
+		sprintf(msg+strlen(msg),"%-4s%-66s%-4s%-20s%-23s%-9s%-7s%-7s%-3s\n",
 			"Ch", "SSID", "UN", "BSSID", "Security", "Siganl(%)", "W-Mode", " ExtCH"," NT");
+	//	sprintf(msg+strlen(msg),"%-4s%-33s%-4s%-20s%-23s%-9s%-7s%-7s%-3s\n",
+	//		"Ch", "SSID", "UN", "BSSID", "Security", "Siganl(%)", "W-Mode", " ExtCH"," NT");
+
 #else
-	sprintf(msg + strlen(msg), "%-4s%-4s%-33s%-20s%-23s%-9s%-7s%-7s%-3s%-8s\n",
+
+    // modify by dimmalex for UTF8 code SSID
+	sprintf(msg + strlen(msg), "%-4s%-4s%-66s%-20s%-23s%-9s%-7s%-7s%-3s%-8s\n",
 		"No", "Ch", "SSID", "BSSID", "Security", "Siganl(%)", "W-Mode", " ExtCH", " NT", " SSID_Len");
+	//sprintf(msg + strlen(msg), "%-4s%-4s%-33s%-20s%-23s%-9s%-7s%-7s%-3s%-8s\n",
+    //    "No", "Ch", "SSID", "BSSID", "Security", "Siganl(%)", "W-Mode", " ExtCH", " NT", " SSID_Len");
+
 #endif /* AIRPLAY_SUPPORT */
 
 
