--- a/src/ndp.c	2024-04-09 23:04:33.062506323 +0800
+++ b/src/ndp.c	2024-04-09 22:57:42.741921150 +0800
@@ -351,7 +351,7 @@
 		return; /* Invalid target */
 
 	inet_ntop(AF_INET6, &req->nd_ns_target, ipbuf, sizeof(ipbuf));
-	syslog(LOG_DEBUG, "Got a NS for %s on %s", ipbuf, iface->name);
+	syslog(LOG_DEBUG, "Got a NS(%d) for %s on %s", ns_is_dad, ipbuf, iface->name);
 
 	odhcpd_get_mac(iface, mac);
 	if (!memcmp(ll->sll_addr, mac, sizeof(mac)))
@@ -363,6 +363,12 @@
 			ping6(&req->nd_ns_target, c);
 	}
 
+	// add by dimmalex to add route to local
+	if ( ns_is_dad )
+	{
+		setup_route( &req->nd_ns_target, iface, true );
+	}
+
 	/* Catch global-addressed NS and answer them manually.
 	 * The kernel won't answer these and cannot route them either. */
 	if (!IN6_IS_ADDR_MULTICAST(&ip6->ip6_dst) &&
